src/test/storage/adaptive_radix_tree_index_test.cpp

```diff
@@ -140,7 +176,62 @@ TEST_F(AdaptiveRadixTreeIndexTest, VectorOfRandomInts) {
       expected_lower += 2;
     }
   }
-  EXPECT_EQ(index->upper_bound({99999}), index->cend());
+
+  for (int search_item = 0; search_item < static_cast<int>(3 * test_size); ++search_item) {
+    if (search_item % 2 == 0 && search_item < static_cast<int>(2 * test_size)) {
+      // all multiples of two within range of $ints should exist
+      EXPECT_NE(index->lower_bound({search_item}), index->cend());
+      continue;
+    }
+
+    // search for elements not existing
+    EXPECT_EQ(index->lower_bound({search_item}), index->upper_bound({search_item}));
+  }
+}
+
+TEST_F(AdaptiveRadixTreeIndexTest, SimpleTest) {
+  std::vector<int> values = {0, 0, 0, 0, 0, 17, 17, 17, 99, std::numeric_limits<int>::max()};
+
+  auto segment = create_dict_segment_by_type<int>(DataType::Int, values);
+  auto index = std::make_shared<AdaptiveRadixTreeIndex>(std::vector<std::shared_ptr<const BaseSegment>>({segment}));
+
+  EXPECT_EQ(*index->cbegin(), 0);
+  EXPECT_EQ(*index->lower_bound({0}), 0);
+  EXPECT_EQ(*index->upper_bound({0}), 5);
+  EXPECT_EQ(*index->lower_bound({17}), 5);
+  EXPECT_EQ(*index->upper_bound({17}), 8);
+  EXPECT_EQ(*index->lower_bound({99}), 8);
+  EXPECT_EQ(*index->upper_bound({99}), 9);
+  EXPECT_EQ(*index->lower_bound({std::numeric_limits<int>::max()}), 9);
+  EXPECT_EQ(index->upper_bound({std::numeric_limits<int>::max()}), index->cend());
+}
+
+/**
+* The following two tests try to test two rather extreme tests that both test the node overflow
+* handling of the ART implementation:
+*   - sparse vector: wide range of values (between 1 and MAX_INT) with large gaps
+*   - dense vector: expenential distribution, rounded to integer values
+**/
+TEST_F(AdaptiveRadixTreeIndexTest, SparseVectorOfRandomInts) {
+  // Increase temporaliy when working on the ART
```

Comment:
`temporaliy` typo. What do you mean with the comment? Someone who is changing the implementation of the art should increase the test_size for the time working on it? Why? Because otherwise the test is not really helpful?

Reply:
Old comment, I'll remove it.
The values should be sufficient (from my understanding) to cover the situations we want to test.

Actually, there is still more to test for the ART which is not covered well yet. But this PR is a starting point.